include/OpenColorIO/OpenColorIO.h

```diff
@@ -634,6 +652,153 @@ class OCIOEXPORT Config
+// getter will return NULL and setter will throw.
+//
+class FileRules
+{
+public:
+    //!cpp:function:: Creates FileRules for a Config. File rules will contain the default rule
+    // using the default role. The default rule can not be removed.
+    static FileRulesRcPtr Create();
+
+    //!cpp:function::
+    FileRulesRcPtr createEditableCopy() const;
+
+    //!cpp:function:: Does include default rule. Result will be at least 1.
+    size_t getNumEntries() const noexcept;
+
+    //!cpp:function:: Get name of the rule.
+    const char * getName(size_t ruleIndex) const;
+
+    //!cpp:function:: Setting pattern will erase regex.
+    const char * getPattern(size_t ruleIndex) const;
+    //!cpp:function::
+    void setPattern(size_t ruleIndex, const char * pattern);
+
+    //!cpp:function:: Setting extension will erase regex.
+    const char * getExtension(size_t ruleIndex) const;
+    //!cpp:function::
+    void setExtension(size_t ruleIndex, const char * extension);
+
+    //!cpp:function:: Setting a regex will erase pattern & extension.
+    const char * getRegex(size_t ruleIndex) const;
+    //!cpp:function::
+    void setRegex(size_t ruleIndex, const char * regex);
+
+    //!cpp:function:: Set the rule's color space (may also be a role).
+    const char * getColorSpace(size_t ruleIndex) const;
+    //!cpp:function::
+    void setColorSpace(size_t ruleIndex, const char * colorSpace);
+
+    //!cpp:function:: Get number of key/value pairs.
+    size_t getNumCustomKeys(size_t ruleIndex) const;
+    //!cpp:function:: Get name of key.
+    const char * getCustomKeyName(size_t ruleIndex, size_t key) const;
+    //!cpp:function:: Get name of value.
+    const char * getCustomKeyValue(size_t ruleIndex, size_t key) const;
+    //!cpp:function:: Adds a key/value or replace value if key exists. Setting a NULL or an
+    // empty value will erase the key.
+    void setCustomKey(size_t ruleIndex, const char * key, const char * value);
+
+    //!cpp:function:: Insert a rule at a given ruleIndex. Rule currently at ruleIndex
+    // will be pushed to index: ruleIndex + 1.
+    // Name must be unique.
+    // - "Default" is a reserved name for the default rule. The default rule is automatically
+    //   added and can't be removed.
+    // - "ColorSpaceNamePathSearch" is also a reserved name (see insertPathSearchRule below).
+    // Will throw if ruleIndex is not less than :cpp:func:`FileRules::getNumEntries`.
+    void insertRule(size_t ruleIndex, const char * name, const char * colorSpace,
+                    const char * pattern, const char * extension);
+    //!cpp:function::
+    void insertRule(size_t ruleIndex, const char * name, const char * colorSpace,
+                    const char * regex);
+    //!cpp:function:: Inserts a rule that uses cpp:func: 'Config:parseColorSpaceFromString'
+    // to search the path for any of the color spaces named in the config (as per OCIO v1).
+    void insertPathSearchRule(size_t ruleIndex);
```

Comment:
Since the "Default" rule is automatically added, can we also add a convenience function like:

`setDefaultRuleColorSpace(const char * colorSpace)`

That would be clearer than having to call this to set the default rule colorspace, since index and name are pre-defined (presumably index means nothing in this case, since it will always fall last):

`insertRule(3, "Default", "my_colorspace");`

Reply:
Good idea!

Reply:
Looks good. Thanks Bernard!